int8 x
int8 y
string orient
string type
int32 weight
int32 size
float64 dis